lib/jeff.ex

```diff
@@ -174,6 +176,79 @@ defmodule Jeff do
     ACU.send_command(acu, address, MFG, params) |> handle_reply()
   end
 
+  @doc """
+  Send file data to a PD
+  """
+  @spec file_transfer(acu(), osdp_address(), binary()) ::
+          Reply.FileTransferStatus.t() | cmd_err()
+  def file_transfer(acu, address, data) when is_binary(data) do
+    file_transfer(acu, address, data, nil)
+  end
+
+  @spec file_transfer(acu(), osdp_address(), binary(), function() | nil) ::
+          Reply.FileTransferStatus.t() | cmd_err()
+  def file_transfer(acu, address, data, progress_callback) when is_binary(data) do
+    total_packets = div(byte_size(data) + @osdp_max_packet_size - 1, @osdp_max_packet_size)
+
+    FileTransfer.command_set(data, @osdp_max_packet_size)
+    |> run_file_transfer(acu, address, progress_callback, 0, total_packets)
+  end
+
+  defp run_file_transfer([cmd | rem], acu, address, progress_callback, packet_num, total_packets) do
+    case ACU.send_command(acu, address, FILETRANSFER, Map.to_list(cmd)) do
+      {:ok, reply} ->
+        handle_file_transfer_reply(
+          reply,
+          rem,
+          acu,
+          address,
+          progress_callback,
+          packet_num,
+          total_packets
+        )
+
+      error ->
+        error
+    end
+  end
+
+  defp run_file_transfer([], _acu, _address, progress_callback, _packet_num, total_packets) do
+    # Handle empty list case
+    if progress_callback do
+      progress_callback.(total_packets, total_packets, 100)
+    end
+
+    %Reply.FileTransferStatus{status: :ok}
+  end
```

Comment:
```suggestion
```
Is this needed? It seems like we should never get to this case because `FileTransfer.adjust_from_reply` should not return an empty list. If we did, I might consider this exceptional since it feels like an empty list here means nothing got updated, but I will believe it has 🤔